tests/infra/partitions.py

```diff
@@ -140,17 +161,22 @@ def isolate_node(
             client_rule["dst"] = other.node_host
             name += f" from node {other.local_node_id}"
 
-        if iptc.easy.has_rule("filter", CCF_IPTABLES_CHAIN, server_rule):
-            iptc.easy.delete_rule("filter", CCF_IPTABLES_CHAIN, server_rule)
+        rules = [
+            server_rule,
+            self.reverse_rule(server_rule),
+            client_rule,
+            self.reverse_rule(client_rule),
+        ]
```

Comment:
The partitions test was failing, and this is the fix. Adding an explanation here, for posterity (and many thanks to @jumaffre for the help spotting this):

Each of these rules becomes a single `DROP` line in `iptables`, which drops _one direction_ of traffic for _one connection_ (potentially only scoped to specific ports). Generally when you do this, you are either a client _or_ a server, and you create 2 symmetric rules; one blocks 'sends', the other blocks 'responses'*, over the 2 known addresses. In our case, we're running both the client _and_ the server, and the 2 rules we created block the 'send' side _but not the respond_. This worked with the previous channels, because all the interesting messages were sent by a channel initiator as 'send's, and since _they_ were blocked we never produced anything to send over the (still available) 'respond' direction.

With the new channel code, an existing connection will be reused if it still exists. So a channel that was initiated by A to talk to B, when A was primary, may later be used by B to talk to A, when B becomes primary. This uses the 'respond' direction of the original connection, though as far as the code is concerned it's always just a 'send' locally (the same as any Raft response, which isn't distinguished from a 'send' by the time it reaches the host).

So this fix is just saying we always block a connection in 2 directions, both the initiator->recipient direction and recipient->initiator. And since we're doing that with 2 client-server address pairs (since either node may have acted as client at some point to initiate the connection), we end up with 4 total rules.

* A connection always has an initiating source, and a receiving destination. For this discussion, over that connection, anything from source to destination is a send, and anything from destination to source is a respond.